### PR TITLE
chore: add 'betterer' to prevent ts strict regressions (#WEBAPP-7144)

### PR DESCRIPTION
**Pull request by PatrykBuniX**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WEBAPP-7144" title="WEBAPP-7144" target="_blank">TaskWEBAPP-7144</a>  Have a job that checks that the number of strict typescript errors is going down
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Our goal is to make our code compiled with TypeScript **strict** mode. As our codebase is pretty big and currently includes about 2300 ts strict errors (null checks, possible undefineds, etc.) It's not possible to just enable the `strict` mode.

We want to prevent us from pushing more typescript `strict` errors.

### Solutions

I've found pretty cool tool called [Betterer](https://phenomnomnominal.github.io/betterer/), which let us keep track of ts compile errors, and saves them as a snapshot. The tool does not allow to increase the number of specified errors, but each time we reduce the number, the snapshot gets updated 😄 

### Setup

Here is the recommended workflow by tool's authors: https://phenomnomnominal.github.io/betterer/docs/workflow. It seems to match our requirements. How it would basically work:

- there is pre-commit husky hook added which will run after `lint-staged`, so it won't allow us to push if we've added any more ts strict errors.
- there is a ci action that will run and make sure our snapshot is in sync with the errors.
- in case we got any merge conflicts in snapshot file, there is a `yarn betterer:conflict` script which should fix it
- in case something bad happens to the snapshot file, we can force its update with `yarn betterer -u`

### Notes

I've never used this tool before, and I need your opinion. I've tested it out locally and works great!

Our codebase is pretty big and `betterer` command can take long time (about 16s for me) on the first run, but thanks to `--cache` option added  next runs should be much faster.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -2,3 +2,4 @@
 . "$(dirname "$0")/_/husky.sh"
 
 npx --no-install lint-staged
+yarn betterer precommit
```
Comment (atomrc) [inline .husky/pre-commit]: How expensive is this command? (in term of time of execution). If it's too long I fear it's going to just have people do the `commit -n` and cancel all checks

Comment (PatrykBuniX) [reply .husky/pre-commit]: It's about 16s for me (m1 mac), It's a lot of files to check. Is it too long? 🤔

Comment (atomrc) [reply .husky/pre-commit]: Oh yeah, that's a bit long for a precommit... :grimacing:

File: .gitignore [hunk 2]
```diff
@@ -16,3 +16,4 @@ CHANGELOG.md
 node_modules
 npm-debug.log
 yarn-error.log
+.betterer.cache
```
Comment (PatrykBuniX) [inline .gitignore:19]: Can this one live here, or we should use individual global ignores for this type of caching things? 🤔

Comment (atomrc) [reply .gitignore:19]: Since we are all going to use it locally, I think it makes sense to have it in the repo :ok_hand:

**Review by atomrc — APPROVED** [review]

**Pull request merged**